From: nbd <nbd@3c298f89-4303-0410-b956-a3cf2f4a3e73>
Date: Sat, 10 Aug 2013 18:35:29 +0000
Subject: x86: Fix CONFIG_X86_GRUB_SERIAL=""

With

    .config:CONFIG_X86_GRUB_SERIAL=""

which (AFAICT) is the way to tell GRUB not to use a serial console, in

    target/linux/x86/image/Makefile:ifneq ($(CONFIG_X86_GRUB_SERIAL),)

$(CONFIG_X86_GRUB_SERIAL) expands to `""' (a literal double double-quote),
making the condition unconditionally false.

This patch fixes the situation by passing CONFIG_X86_GRUB_SERIAL through
qstrip before testing.

Signed-off-by: Tamas TEVESZ <ice@extreme.hu>

diff --git a/target/linux/x86/image/Makefile b/target/linux/x86/image/Makefile
index 6e2bd54..2e73519 100644
--- a/target/linux/x86/image/Makefile
+++ b/target/linux/x86/image/Makefile
@@ -27,8 +27,10 @@ ifneq ($(strip $(foreach subtarget,$(USE_ATKBD),$(CONFIG_TARGET_x86_$(subtarget)
   GRUB2_MODULES += at_keyboard
 endif
 
-ifneq ($(CONFIG_X86_GRUB_SERIAL),)
-  GRUB_CONSOLE_CMDLINE += console=$(call qstrip,$(CONFIG_X86_GRUB_SERIAL)),$(CONFIG_X86_GRUB_BAUDRATE)n8
+GRUB_SERIAL:=$(call qstrip,$(CONFIG_X86_GRUB_SERIAL))
+
+ifneq ($(GRUB_SERIAL),)
+  GRUB_CONSOLE_CMDLINE += console=$(GRUB_SERIAL),$(CONFIG_X86_GRUB_BAUDRATE)n8
   GRUB_SERIAL_CONFIG := serial --unit=0 --speed=$(CONFIG_X86_GRUB_BAUDRATE) --word=8 --parity=no --stop=1
   GRUB_TERMINALS += serial
 endif
